File: src/main/java/susc/s01/IOHandler/FileMap.java
package susc.s01.IOHandler;

import susc.s01.Data.Log.JoinLog.PlayerLog;
import susc.s01.SUSC_craft;

import java.io.File;

public enum FileMap {
    JOIN_LOG(new File(SUSC_craft.getPlugin(SUSC_craft.class).getDataFolder(), "JoinLog.json"));

    private final File filePathInstance;

    FileMap(File filePathInstance) {
        this.filePathInstance = filePathInstance;
    }

    public File getFilePathInstance() {
        return filePathInstance;
    }

    public String getFileName() {
        return filePathInstance.getName();
    }
}

File: src/main/java/susc/s01/IOHandler/FileMap.kt
package susc.s01.IOHandler

import susc.s01.Data.Log.JoinLog.PlayerLog
import susc.s01.SUSC_craft
import java.io.File

enum class FileMap(val filePathInstance: File, val fileInstanceType: Class<*>) {
    JOIN_LOG(
        File(SUSC_craft.getPlugin<SUSC_craft>(SUSC_craft::class.java).getDataFolder(), "JoinLog.json"),
        PlayerLog::class.java
    );

    val fileName: String
        get() = filePathInstance.name
}
